harness/determined/common/storage/base.py

```diff
@@ -2,11 +2,17 @@
 import contextlib
 import os
 import pathlib
-from typing import Any, Callable, Dict, Iterator, Optional, Union
+from typing import Any, Callable, Dict, Iterator, Optional, Set, Union
+
+# Paths should be a sorted list of paths relative to the checkpoint root that indicate what paths
+# should be uploaded.  A directory should always appear in Paths if any subpath under that directory
+# appears in Paths.
+Paths = Set[str]
```

Comment:
hm, I didn't mean to make `Paths` (the type) a set, I meant in that one instance to convert it to a set before using it.

But I suppose this works, and since it's an internal API anyway, I think its a fine change.

But you need to change the docstring which still says "sorted list"

Reply:
yeah, initially, I only changed it to set in that one place... and then I decided to change Paths definition to set since there was no reason not to. good catch on the comment!